### PR TITLE
Deploy release builds

### PR DESCRIPTION
**Pull request by arielshaqed**
Allows releasing a build and thus finishing #2081 :-)

* Add "release" profile that also creates Javadoc and sources artifacts.
* Remove "SNAPSHOT" so that we can publish.

iolakefs-1015 and iolakefs-1016 staging repositories on Nexus for your edification.

To release:

```sh
ariels@redqueen:~/dev/lakeFS/clients/hadoopfs$ mvn -Prelease -Passembly clean deploy
ariels@redqueen:~/dev/lakeFS/clients/hadoopfs$ mvn -Prelease clean deploy
```

### CODE REVIEW

**Comment by arielshaqed** [comment]
Right now use -0.1.0-rc.0 works beautifully.  Changing to 0.1.0 to release -- @talSofer please review and block if something's bad.

File: docs/integrations/spark.md [hunk 1]
```diff
@@ -216,10 +216,10 @@ FileSystem.
 #### Load the FileSystem JARs
 
 Add the package `io.lakefs:hadoop-lakefs-assembly:<VERSION>` to your Spark job.  Right now
-this is a snapshot, so you add
+this is version 0.1.0, so add:
 
 ```
---repositories https://s01.oss.sonatype.org/content/repositories/snapshots/  --packages io.lakefs:hadoop-lakefs-assembly:0.1.0-RC.0-SNAPSHOT
+--repositories --packages io.lakefs:hadoop-lakefs-assembly:0.1.0
```
Comment (talSofer) [inline docs/integrations/spark.md:222]: do we still need the "repositories" flag?

**Review by talSofer — APPROVED** [review]
Can you document the lakefs filesystem release process in dev please?

File: docs/integrations/spark.md (hunk 1 above)
Comment (arielshaqed) [reply docs/integrations/spark.md:222]: Yikes, no!

**Comment by arielshaqed** [comment]
> Can you document the lakefs filesystem release process in dev please?

Will do!

**Review by arielshaqed — COMMENTED** [review]
Thanks!

Pulling...

**Pull request merged**